src/main/java/se/llbit/json/JsonParser.java

```diff
@@ -323,6 +323,38 @@ private JsonMember parseMember() throws IOException, SyntaxError {
     return null;
   }
 
+  private String parseObjectKey() throws IOException, SyntaxError {
+    if (in.peek() == Literal.QUOTE_MARK) {
+      return parseString();
+    } else if (isValidInKey((char) in.peek())) {
+      StringBuilder sb = new StringBuilder();
+      while (true) {
+        int next = in.peek();
+        if (next == EOF) {
+          String result = sb.toString();
+          if (result.isEmpty()) {
+            throw new SyntaxError("unexpected end of input");
+          }
+          return result;
+        } else if (!isValidInKey((char) next)) {
+          String result = sb.toString();
+          if (result.isEmpty()) {
+            throw new SyntaxError(String.format("unexpected character '%c'", (char) next));
+          }
+          return result;
+        }
+        sb.append((char) next);
+        in.pop();
+      }
+    } else {
+      throw new SyntaxError(String.format("unexpected character '%c'", (char) in.peek()));
+    }
+  }
+
+  private boolean isValidInKey(char c) {
+    return Character.isAlphabetic(c);
```

Comment:
Identifiers in JS may actually contain numbers (and pretty much anything else), but just not start with them: https://stackoverflow.com/questions/1661197/what-characters-are-valid-for-javascript-variable-names